### ISSUE TITLE
Use since argument on journalctl with very old date doesn't return the logs

### ISSUE THREAD

**Issue by lazize** [systemd/systemd#6290]
I am not sure if it is a bug, I tried to search documentation about date limitation, or range, but couldn't find anything.

### Submission type
 - Bug report

### systemd version the issue has been seen with

```shell
:~$ systemd --version
systemd 229
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ -LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN
```

### Used distribution

```shell
:~$ cat /etc/*release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.2 LTS"
NAME="Ubuntu"
VERSION="16.04.2 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.2 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

### In case of bug report: Expected behaviour you didn't see

As you can see below, using a "newer" date it returns `650` lines, but using an older date it doesn't return anything.
```shell
:~$ sudo journalctl --no-pager -au docker --since "2000-01-01" | wc -l
650

:~$ sudo journalctl --no-pager -au docker --since "1900-01-01"
-- No entries --
```

### In case of bug report: Unexpected behaviour you saw

> See it above please

### In case of bug report: Steps to reproduce the problem

> See it above please

**Comment by fsateler** [comment]
Turns out `parse_timestamp` for some reason rejects time from before the epoch:

```
% ./journalctl --since=1969-12-31 | head -0
Failed to parse timestamp: 1969-12-31
% ./journalctl --since=1970-01-01 | head -0
% 
```

It seems to me the check [`x<0`](https://github.com/systemd/systemd/blob/master/src/basic/time-util.c#L850) should be changed to `x == (time_t)-1`.

**Comment by fsateler** [comment]
No, that's not enough. usec_t is unsigned so negative offsets are larger than USEC_TIMESTAMP_FORMATTABLE_MAX

**Comment by yuwata** [comment]
`journalctl --since=-100years` works fine. See [time-util.c#L870](https://github.com/systemd/systemd/blob/master/src/basic/time-util.c#L870).
So, how about modifying [time-util.c#L851](https://github.com/systemd/systemd/blob/master/src/basic/time-util.c#L851) as, for example, the following.
```
x = mktime_or_timegm(&tm, utc);
if (x < 0)
        x = 0;
```

**Comment by yuwata** [comment]
I've created a PR for this bug. Please take a look.

**Comment by poettering** [comment]
I don#t like the fix #6297, as it creates an ambiguity... I think we should cleanly refuse parsing such old dates for our purposes. Reopening

**Comment by yuwata** [comment]
OK. I will create another PR.
